packages/react-native-bpk-component-chip/src/BpkChipWrapper.js

```diff
@@ -101,59 +126,74 @@ const BpkChipWrapper = (props: Props) => {
     label,
     selected,
     style,
+    type,
     theme,
     ...rest
   } = props;
 
+  const userStyle = [style];
   const innerStyle = [styles.inner];
   const textStyle = [styles.text];
   const iconStyle = [styles.icon];
 
+  if (type === CHIP_TYPES.outline) {
+    userStyle.push({ elevation: 0 });
+    innerStyle.push(styles.innerOutline);
+    textStyle.push(styles.textOutline);
+    iconStyle.push(styles.iconOutline);
+  }
+
   if (selected) {
     innerStyle.push(styles.innerSelected);
     textStyle.push(styles.textSelected);
+    if (type === CHIP_TYPES.outline) {
+      innerStyle.push(styles.innerOutlineSelected);
+    }
   }
 
   if (disabled) {
     innerStyle.push(styles.innerDisabled);
     textStyle.push(styles.textDisabled);
     iconStyle.push(styles.iconDisabled);
+    if (type === CHIP_TYPES.outline) {
+      innerStyle.push(styles.innerOutlineDisabled);
+      textStyle.push(styles.textOutlineDisabled);
+      iconStyle.push(styles.iconOutlineDisabled);
+    }
   }
 
   const themeAttributes = {
     ...getThemeAttributes(REQUIRED_THEME_ATTRIBUTES, theme),
-    ...getThemeAttributes(OPTIONAL_THEME_ATTRIBUTES, theme),
   };
 
-  if (themeAttributes) {
-    if (themeAttributes.colorGray700 && themeAttributes.colorGray500) {
-      textStyle.push({
-        color: themeAttributes.colorGray700,
-      });
-      iconStyle.push({
-        color: themeAttributes.colorGray500,
-      });
-    }
-    if (
-      selected &&
-      themeAttributes.chipSelectedBackgroundColor &&
-      themeAttributes.chipSelectedTextColor
-    ) {
-      innerStyle.push({
-        backgroundColor: themeAttributes.chipSelectedBackgroundColor,
-      });
-      textStyle.push({
-        color: themeAttributes.chipSelectedTextColor,
-      });
-    }
-    if (disabled && themeAttributes.colorGray300) {
-      textStyle.push({
-        color: themeAttributes.colorGray300,
-      });
-      iconStyle.push({
-        color: themeAttributes.colorGray300,
-      });
-    }
+  const shouldApplyTheme = themeAttributes && selected && !disabled;
+  const shouldApplyPrimaryTheme =
+    shouldApplyTheme &&
+    type === CHIP_TYPES.primary &&
+    themeAttributes.chipSelectedBackgroundColor &&
+    themeAttributes.chipSelectedTextColor;
+
+  const shouldApplyOutlineTheme =
+    shouldApplyTheme &&
+    type === CHIP_TYPES.outline &&
+    themeAttributes.chipOutlineSelectedBackgroundColor &&
+    themeAttributes.chipOutlineSelectedTextColor;
+
+  if (shouldApplyPrimaryTheme) {
+    innerStyle.push({
+      backgroundColor: themeAttributes.chipSelectedBackgroundColor,
+    });
+    textStyle.push({
+      color: themeAttributes.chipSelectedTextColor,
+    });
+  }
+  if (shouldApplyOutlineTheme) {
+    innerStyle.push({
+      backgroundColor: themeAttributes.chipOutlineSelectedBackgroundColor,
+    });
+    textStyle.push({
+      color: themeAttributes.chipOutlineSelectedTextColor,
+    });
```

Comment:
This whole section of refactoring is a really good example of readability, the code is really easy to follow. Nice one 😄

Reply:
This is thanks to @georgegillams who suggested it and provided it 😄